ui/src/tool-panel/location-track/split-store.ts

```diff
@@ -439,29 +478,47 @@ function updateAutogeneratedSplitNames(
     firstSplit: FirstSplitTargetCandidate;
     splits: SplitTargetCandidate[];
 } {
-    const firstSwitch = trackSwitches.find((sw) => sw.switchId == startSwitchId);
+    const firstSwitch = trackSwitches.find(
+        (sw) =>
+            startSplitPoint?.type == 'switchSplitPoint' && sw.switchId == startSplitPoint.switchId,
+    );
+    const firstSplitEndSplitPoint = splits.length > 0 ? expectDefined(splits[0]).splitPoint : endSplitPoint;
+    const firstSplitEndSwitch = trackSwitches.find(
+        (sw) =>
+            firstSplitEndSplitPoint?.type == 'switchSplitPoint' &&
+            sw.switchId == firstSplitEndSplitPoint.switchId,
+    );
     const newFirstSplit =
         firstSplit.hasAutogeneratedName && splits.length > 0
             ? {
                   ...firstSplit,
-                  ...getNameForTarget(firstSwitch, expectDefined(splits[0]).switch, trackNumber),
+                  ...getNameForTarget(firstSwitch, firstSplitEndSwitch, trackNumber),
               }
             : firstSplit;
 
-    const endSwitch = trackSwitches.find((sw) => sw.switchId == endSwitchId);
-    const newSplits = splits.map((split, index) =>
-        split.hasAutogeneratedName
+    const endSwitch = trackSwitches.find(
+        (sw) => endSplitPoint?.type == 'switchSplitPoint' && sw.switchId == endSplitPoint.switchId,
+    );
+    const newSplits = splits.map((split, index) => {
+        const splitStartSwitch = trackSwitches.find(
+            (sw) =>
+                split.splitPoint.type == 'switchSplitPoint' &&
+                split.splitPoint.switchId == sw.switchId,
+        );
+        const nextSplit = index + 1 < splits.length ? splits[index + 1] : undefined;
```

Comment:
Tässähän periaatteessa ei ole tarvetta tehdä tarkasteluja, vaan `splits[index +1]` palauttaa `undefined`:in ohi-indeksoidessa. Eipä tuosta tosin haittaakaan ole, mutta kumminkin.